File: e2e/playwright-python/interactions/pages/language_switcher_dropdown.py
from playwright.sync_api import Page, Locator


class LanguageSwitcherDropdown:
    def __init__(self, page: Page):
        self.dropdown: Locator = page.get_by_test_id("language-switcher")
        self.dropdown_value: Locator = self.dropdown.locator("span.p-dropdown-label")
        self.items: Locator = page.locator(".p-dropdown-item")

File: e2e/cucumber-playwright-typescript/interactions/pages/LanguageSwitcherDropdown.ts
import { Locator, Page } from '@playwright/test';

export class LanguageSwitcherDropdown {
  readonly dropdown: Locator;
  readonly dropdownValue: Locator;
  readonly items: Locator;

  constructor(page: Page) {
    this.dropdown = page.getByTestId('language-switcher');
    this.dropdownValue = this.dropdown.locator('span.p-dropdown-label');
    this.items = page.locator('.p-dropdown-item');
  }
}
